metashare/api/models.py

```diff
@@ -342,27 +352,29 @@ def notify_has_url(self):
         from .serializers import ScratchOrgSerializer
 
         payload = ScratchOrgSerializer(self).data
-        message = {"type": "SCRATCH_ORG_PROVISIONED", "payload": payload}
+        message = {"type": "SCRATCH_ORG_PROVISION", "payload": payload}
         async_to_sync(push.push_message_about_instance)(self, message)
 
     # def notify_refreshing_changes(self):
     #     from .serializers import ScratchOrgSerializer
 
     #     payload = ScratchOrgSerializer(self).data
-    #     message = {"type": "SCRATCH_ORG_UPDATED", "payload": payload}
+    #     message = {"type": "SCRATCH_ORG_UPDATE", "payload": payload}
     #     async_to_sync(push.push_message_about_instance)(self, message)
 
     def notify_deleted(self):
         from .serializers import ScratchOrgSerializer
 
         payload = ScratchOrgSerializer(self).data
-        message = {"type": "SCRATCH_ORG_DELETED", "payload": payload}
+        message = {"type": "SCRATCH_ORG_DELETE", "payload": payload}
         async_to_sync(push.push_message_about_instance)(self, message)
 
 
 @receiver(user_logged_in)
 def user_logged_in_handler(sender, *, user, **kwargs):
-    user.refresh_repositories()
+    from .jobs import refresh_github_repositories_for_user_job
+
+    refresh_github_repositories_for_user_job.delay(user)
```

Comment:
Problem: On initial page-load, the front end has no way of knowing whether or not the user just logged in, and therefore whether or not there's a current refresh happening in the background. There's a delay of a few seconds before the job starts, and this handler fires too early to be useful (i.e. before the front end subscribes to the user WS channel).

Proposal: In addition to triggering an automatic refresh after every login, also trigger an automatic refresh on every page load _if the user has no repos_. That way on every page load, if there are no repos the front end can assume there's a background refresh happening, and wait for it to finish before displaying the app.

Reply:
@wlonk (cc @davisagli) I'd be happy for your approval or feedback on the above -- or to discuss it further on a call.